### PR TITLE
Add Rounded Polygon node

### PR DESCRIPTION
**Pull request by RemyUnity**
Add a Rounded Polygon node.

[Katana link](https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=sg%2FRoundedPoly&automation-tools_branch=master&unity_branch=trunk&sort=1-asc)

### CODE REVIEW

**Comment by Kink3d** [comment]
Yea this is still going in, just havent had the time to make the needed changes. Will wait until after 18.3 branch off.

**Comment by sebastienlagarde** [comment]
Is this PR still active ?

**Comment by RemyUnity** [comment]
@Kink3d What are the needed changes ? Moving the code to not use the deprecated custom node api I guess ?
If you think this is not usefull maybe we can just deleted this PR, or I even I can do the migration.
I just feel sad that this think takes so long for a "simple" node ...

File: com.unity.shadergraph/Editor/Data/Nodes/Procedural/Shape/RoundedPolygon.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
+﻿using UnityEngine;
+using UnityEditor.ShaderGraph;
+using System.Reflection;
+
+[Title("Procedural", "Shape", "Rounded Polygon")]
+public class RoundedPolygon : CodeFunctionNode
```
Comment (pbbastian) [inline com.unity.shadergraph/Editor/Data/Nodes/Procedural/Shape/RoundedPolygon.cs]: Put this into the `UnityEditor.ShaderGraph` namespace

File: com.unity.shadergraph/Editor/Data/Nodes/Procedural/Shape/RoundedPolygon.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
+﻿using UnityEngine;
+using UnityEditor.ShaderGraph;
```
Comment (pbbastian) [inline com.unity.shadergraph/Editor/Data/Nodes/Procedural/Shape/RoundedPolygon.cs]: No longer needed after the other requested change

Comment (RemyUnity) [reply com.unity.shadergraph/Editor/Data/Nodes/Procedural/Shape/RoundedPolygon.cs]: Apparently UnityEngine is still needed after including the classe to the ShaderGraph namespace. For the Vector2 used in the first input slot.

Comment (pbbastian) [reply com.unity.shadergraph/Editor/Data/Nodes/Procedural/Shape/RoundedPolygon.cs]: Yep, only line 2 should be removed :)

File: com.unity.shadergraph/Editor/Data/Nodes/Procedural/Shape/RoundedPolygon.cs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        polaruv.x += PI;
+
+        polaruv.x = fmod( polaruv.x + halfAngle, fullAngle );
+        polaruv.x = abs(polaruv.x - halfAngle);
+
+        UV = {precision}2( cos(polaruv.x), sin(polaruv.x) ) * polaruv.y;
+
+        Out = UV.x;
+
+        if ( ( halfAngle - polaruv.x ) < chamferAngle )
```
Comment (Kink3d) [inline com.unity.shadergraph/Editor/Data/Nodes/Procedural/Shape/RoundedPolygon.cs]: Remove branch

File: com.unity.shadergraph/Editor/Data/Nodes/Procedural/Shape/RoundedPolygon.cs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            [Slot(2, Binding.None, 0.5f, 0, 0, 0)] Vector1 Height,
+            [Slot(3, Binding.None, 5f, 0, 0, 0)] Vector1 Sides,
+            [Slot(4, Binding.None, 0.3f, 0, 0, 0)] Vector1 Roundness,
+            [Slot(5, Binding.None)] out Vector1 Out)
+        {
+            return
+                @"
+{
+	UV = UV * 2. + {precision}2(-1.,-1.);
+    UV.x = (Width==0)? 0xFFFFFF : UV.x / Width;
```
[the remark this answers is not in the document]
Comment (RemyUnity) [reply com.unity.shadergraph/Editor/Data/Nodes/Procedural/Shape/RoundedPolygon.cs]: I didn't find a way to get rid of this branch that doesn't involve increasing the cost of the shader, do you have any clue ?
How to have the "Width" (and Height) value not 0, but keep it's sign ?

**Comment by RemyUnity** [comment]
Making those last modifications, I wanted to be as close as possible to the behaviour of the "Polygon Shape" node.

But how should the shape size adapt to the roundness value ?

1) Scale to fit to the Width/Height:
RoundedPoly_01

2) Dont change and get "smaller":
RoundedPoly_02

I guess I could also make it a toggle, or even a controllable input. What do you think is the best ?

**Comment by RemyUnity** [comment]
I guess that you'll want this to be included to the test scenes.

Because this PR is quite old, I'm not very comfy on how to handle that :
- Should I want for this PR to get merged and create a new one to add the test ?
- Merge master in this PR, and add the test ?

**Comment by pbbastian** [comment]
I'd say merge master in this PR and add the test :) Then we also decrease the risk of merge skew, since the PR is fairly old

**Comment by RemyUnity** [comment]
Added it to the procedural nodes test scene:
new test

Katana is green.

**Comment by pbbastian** [comment]
It looks like something went wrong in the merge, given how many files are conflicting now.

**Comment by alindmanUnity** [comment]
Going to rebuild this branch and port the node code over to a clean PR
https://favro.com/organization/c564ede4ed3337f7b17986b6/d06b8de3e40da6d340b5b063?card=Uni-94299

**Comment by alindmanUnity** [comment]
PR rebuilt here #4532

**Pull request closed without merging**